server/src/metrics/transform/transforms.service.ts

```diff
@@ -0,0 +1,105 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { Metric as MetricEntity } from "../../database/entities/metric.entity";
+import { PrometheusQueryService } from "../prometheus-query/query.service";
+import { SeriesEntity } from "../metrics.service";
+
+type Current = {
+  id: number;
+  timestamp: string;
+  data: string;
+};
+
+@Injectable()
+export class TransformsService {
+  private readonly logger = new Logger(TransformsService.name);
+
+  constructor(
+    @InjectRepository(MetricEntity)
+    private metricRepository: Repository<MetricEntity>,
+    private prometheusQuery: PrometheusQueryService,
+  ) {}
+
+  // Get the current value for a metric
+  async getSumTotal(name: string): Promise<Current> {
+    const prometheusQuery = await this.prometheusQuery.get(name);
+
+    const query = this.metricRepository
+      .createQueryBuilder("m")
+      .where("m.prometheusQueryId = :prometheusQuery", {
+        prometheusQuery: prometheusQuery.id,
+      })
+      .orderBy("m.timestamp", "DESC");
+
+    const values: MetricEntity[] = await query.getMany();
+
+    if (!values) {
+      return null;
+    }
+
+    const sum = values.reduce(
+      (accumulator, item) => accumulator + Number(item.data),
+      0,
+    );
+
+    const sumTotal = {
+      id: Math.floor(Math.random() * 900000) + 100000,
+      timestamp: new Date().toISOString(),
+      data: sum.toString(),
+    };
+
+    return sumTotal;
+  }
+
+  async getSeriesSumTotal(
+    name: string,
+    from: Date,
+    to: Date,
+  ): Promise<SeriesEntity[] | null> {
+    const prometheusQuery = await this.prometheusQuery.get(name);
+
+    const query = this.metricRepository
+      .createQueryBuilder("m")
+      .where("m.prometheusQueryId = :prometheusQuery", {
+        prometheusQuery: prometheusQuery.id,
+      })
+      .orderBy("m.timestamp", "DESC");
```

Comment:
Add a comment that this query returns *every data point* for that given metric, being a year or 10.